cmake/ExternalProjectHelper.cmake

```diff
@@ -29,7 +29,7 @@ function(nrn_initialize_submodule path)
   endif()
   message(STATUS "Sub-module : missing ${path} : running git submodule update --init")
   execute_process(
-    COMMAND ${GIT_EXECUTABLE} submodule update --init -- ${path}
+    COMMAND ${GIT_EXECUTABLE} submodule update --recursive --init -- ${path}
```

Comment:
Why `--recursive` still? Just for nanobind? Can we try to do that instead in `.gitmodules`? Otherwise we pull the big string and slurp in all of nmodl's submodules ⇒ lots of sources!

Reply:
Done

Reply:
Surprisingly this is not working as expected.

Reply:
So `fetchRecurseSumodules` is only for `fetch` and `pull`.
`git submodule update` is doing a `checkout`.

So I will clone the `git` repository and read the code to see if there is an hidden variable to set that.

Stay in touch.

Reply:
So in `git` there is 2 different things, core module written in `C`, and helpers previously written in `perl` but today rewrite in `C` but not really integrated to the core.

`git submodule` command is one of those previous helper and so the integration with the core is not perfect. It does not take care of `options` for example.

Should I tried to patch git to have a better consideration of options?

For now, I will continue to `update --init --recursive` but still no `--depth=1`. 

Don't be afraid, I will not merge like that, just trying to continue the debugging of windows/symbol/cmake things with tmate.

Reply:
OK :)

Thank you for looking in depth there!

Reply:
@alkino: just one more point here about the need for `--recursive`: I think the only reason we need `--init --recursive` here is because `nanobind` itself has [a submodule]( https://github.com/wjakob/nanobind/blob/bd6256570235f38a2c7a2193f2469cde00c7bca5/.gitmodules#L3):

```console
[submodule "ext/robin_map"]
	path = ext/robin_map
	url = https://github.com/Tessil/robin-map
```

IIRC, we were not preferring  `--recursive` because submodules might end-up cloning unnecessarily their own sub-submodules.

If we still want to avoid `--recursive`, one option could to add to extra argument to `nrn_add_external_project()` which decides if `--recursive` should be used or not.

I quickly checked the sub-modules cloned in nrn. If I am not mistaken, there is not any specific submodule that is big/heavy (like nmodl, IIRC). So I don't have a strong opinion here!